### PR TITLE
[bitnami/kafka] Fix Kafka log/logs dir config

### LINKED ISSUE #20077: Unable to mount hostpath to bitname/kafka pods
### Name and Version

bitnami/kafka version 25.1.12 

### What architecture are you using?

None

### What steps will reproduce the bug?
```
 helm upgrade --install kafkalog1 bitnami/kafka -n example --set persistence.size=250Gi -f kafka-rest/kafka.yaml -f kafka-rest/metrics.yaml
```
### Are you using any custom parameters or values?
```
 helm upgrade --install kafkalog1 bitnami/kafka -n example --set persistence.size=250Gi -f kafka-rest/kafka.yaml -f kafka-rest/metrics.yaml
```

### What is the expected behavior?

I'm adding this content to kafka.yaml but I see the hostpath configuration is ignored.
```yaml
volumeMounts:
  - mountPath: "/log"
    name: temporal

volumes:
- name: temporal
  hostPath:
    path: "/log"
    type: DirectoryOrCreate
```
### What do you see instead?

The hostpath is not mounted.

### Additional information

The hostpath is not mounted.

I have tried these things 
```
helm3 upgrade --install clusterkafkalog1 bitnami-full-index/kafka -n example --set volumes.temporal.hostPath.path=/logs --set persistence.size=10Gi,volumePermissions.enabled=true,persistence.enabled=true -f kafka-rest-log-1a/kafka.yaml -f kafka-rest-log-1a/metrics.yaml --version 25.1.12 
```

### PR DESCRIPTION
**Pull request by juan131**
### Description of the change

We're using controller data persistence configuration for broker pods which is incorrect since we're defining different parameters for each Kafka node type.

Also, we're currently mounting a `logs` volume in Kafka container (using an emptyDir or PVClaimTemplate) but we're not configuring the `logs.dir` property that's used on default log4j configuration, see:

- https://github.com/apache/kafka/blob/trunk/config/log4j.properties 

### Benefits

Chart to be consistent with its exposed parameters.

### Possible drawbacks

None

### Applicable issues

- fixes #20077

### Additional information

N/A

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/kafka/templates/_helpers.tpl [hunk 1]
```diff
@@ -598,7 +598,7 @@ Returns the controller quorum voters based on the number of controller-eligible
 Section of the server.properties configmap shared by both controller-eligible and broker nodes
 */}}
 {{- define "kafka.commonConfig" -}}
-log.dir={{ printf "%s/data" .Values.controller.persistence.mountPath }}
+log.dir={{ .Values.controller.logPersistence.mountPath }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/kafka/templates/_helpers.tpl:601]: Good catch! Fixed at https://github.com/bitnami/charts/pull/20476/commits/e2e55df36bf19a4f816802fee7ce55317ca7b5fa

File: bitnami/kafka/templates/broker/configmap.yaml [hunk 2]
```diff
@@ -40,6 +40,8 @@ data:
     inter.broker.protocol.version={{ default (regexFind "^[0-9].[0-9]+" .Chart.AppVersion) .Values.interBrokerProtocolVersion }}
     {{- end }}
     {{- end }}
+    log.dir={{ printf "%s/data" .Values.broker.persistence.mountPath }}
+    logs.dir={{ .Values.broker.logPersistence.mountPath }}
```
Comment (migruiz4) [inline bitnami/kafka/templates/broker/configmap.yaml:44]: Could you please add a comment to help differentiate between both settings?

```suggestion
    # Kafka data logs directory
    log.dir={{ printf "%s/data" .Values.broker.persistence.mountPath }}
    # Kafka application logs directory
    logs.dir={{ .Values.broker.logPersistence.mountPath }}
```

Comment (juan131) [reply bitnami/kafka/templates/broker/configmap.yaml]: done at https://github.com/bitnami/charts/pull/20476/commits/e670035011b154d8cc9ce843fb095b787ca49b26

**Review by migruiz4 — APPROVED** [review]
LGTM!

**Pull request merged**